WalletWasabi.Fluent/State/StateMachine.cs

```diff
@@ -95,7 +98,7 @@ public class StateContext
 		private StateContext? _parent;
 		private List<Action> _entryActions;
 		private List<Action> _exitActions;
-		private List<Action> _onProcessActions;
+		private Dictionary<TTrigger, List<Action>?> _triggerActions;
```

Comment:
Please consider avoiding null when they are not strictly necessary. In this case there is no reason to add a `null` when it would make more sense to add an empty list.

```suggestion
		private Dictionary<TTrigger, List<Action>> _triggerActions;
```

Reply:
remember ? does not change the MSIL generated by the compiler its 100% just an annotation. So if you can still get an NRE then removing ? doesnt help you.

Dictionary is a special case where you cannot guarentee it wont be null unfortunately.